### PR TITLE
Add test for GenericPublisher/Subscriber

### PR DESCRIPTION
**Pull request by h-suzuki-isp**
Please, refer to  https://github.com/ros2/rclcpp/pull/2448.

Because I added tracepoint for GenericPublisher and GenericSubscriber, I add some tests to test_tracetools.
I have added the following tests.
- src/
  - test_generic_ping.cpp
  - test_generic_pong.cpp
- test/
  - test_generic_pub_sub.py
  - test_generic_subscription.py

action-ros-ci-repos-override: https://gist.githubusercontent.com/christophebedard/9a37e393935913add5ad490aaee26e7f/raw/194c464b68cd138e887f041f50f42c0cbe3f4710/ros2.repos

### CODE REVIEW

**Comment by christophebedard** [comment]
The `Rpr__ros2_tracing__ubuntu_noble_amd64` job failure is expected due to the transition to Ubuntu Noble.

The reason why the `ros2_tracing / test (ubuntu-22.04, rolling, source, instr-enabled, tp-included)` job fails is because only this PR branch gets built; your `rclcpp`/`rcl`/`rmw_*` branches are not used. I've added a repos file to this PR description so that it runs with your other changes. Can you do `git commit --amend --no-edit` and then force push so that CI runs again?

**Comment by h-suzuki-isp** [comment]
@christophebedard 
Thank you for your help.
As a result of running CI again, the test pass successfully.

**Comment by christophebedard** [comment]
It looks good to me, but I'm just trying to run the tests with the other rmw implementations too. For some reason I can't get `RMW_IMPLEMENTATION=rmw_connextdds` to work.

**Comment by christophebedard** [comment]
> It looks good to me, but I'm just trying to run the tests with the other rmw implementations too. For some reason I can't get `RMW_IMPLEMENTATION=rmw_connextdds` to work.

Oops, I didn't have `rti-connext-dds-6.0.1` installed on my system, that's probably why.

File: test_tracetools/test/test_generic_pub_sub.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+                tp.rcl_publisher_init,
+                tp.rmw_publish,
+                tp.rcl_publish,
+                tp.rclcpp_publish,
+                tp.rmw_subscription_init,
+                tp.rcl_subscription_init,
+                tp.rclcpp_subscription_init,
+                tp.rclcpp_subscription_callback_added,
+                tp.callback_start,
+                tp.callback_end,
```
Comment (christophebedard) [inline test_tracetools/test/test_generic_pub_sub.py:41]: Is there any reason why the `rmw_take` tracepoint is missing from the test? The `rmw_*` PRs add an `rmw_take` tracepoint for serialized takes, so it should ideally be tested here.

File: test_tracetools/test/test_generic_subscription.py [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+from tracetools_trace.tools.lttng import is_lttng_installed
+
+
+@unittest.skipIf(not is_lttng_installed(minimum_version='2.9.0'), 'LTTng is required')
+class TestSubscription(TraceTestCase):
+
+    def __init__(self, *args) -> None:
+        super().__init__(
+            *args,
+            session_name_prefix='session-test-subscription',
```
Comment (christophebedard) [inline test_tracetools/test/test_generic_subscription.py:31]:
```suggestion
            session_name_prefix='session-test-generic-subscription',
```

File: test_tracetools/test/test_generic_pub_sub.py (hunk 1 above)
Comment (h-suzuki-isp) [reply test_tracetools/test/test_generic_pub_sub.py]: @christophebedard 
Sorry, I implemented `test_generic_*.py` with reference to it before https://github.com/ros2/ros2_tracing/commit/dadd0e9a1a9f4b8e224a0dbfd78b7f96e48bc6af.
I didn't notice that it had been updated.
I fixed `test_generic_pub_sub.py` in https://github.com/ros2/ros2_tracing/pull/97/commits/983cc237b50739e314a045da6c9c23886323d1de.
Also, `test_generic_subscription.py` was updated, so I fixed in https://github.com/ros2/ros2_tracing/pull/97/commits/9ddf39e1ac4497326ad5ff071b42feea352e7995.

Comment (christophebedard) [reply test_tracetools/test/test_generic_pub_sub.py]: Thank you for updating the new tests!

**Review by christophebedard — APPROVED** [review]
Thanks for iterating!

**Pull request merged**